File: app/src/main/java/com/some/mvvmdemo/mvvm/ShareDataVM.java
package com.some.mvvmdemo.mvvm;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.MutableLiveData;

public class ShareDataVM extends AndroidViewModel {

    private MutableLiveData<String> mLiveData = new MutableLiveData<>();

    public ShareDataVM(@NonNull Application application) {
        super(application);
    }

    public MutableLiveData<String> getLiveData() {
        return mLiveData;
    }
}

File: app/src/main/java/com/some/mvvmdemo/mvvm/ShareDataVM.kt
package com.some.mvvmdemo.mvvm

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.launch

class ShareDataVM(application: Application) : AndroidViewModel(application) {

    val liveData = MutableLiveData<String>()

    fun request(){
        viewModelScope.launch{

        }
    }

}
